File: web/src/api/types/logical-codebase-registration.ts
// 逻辑代码库登记 API 的后端 snake_case DTO 投影。
export type RegistrationPreflightRequest = {
  aggregate_root: string;
  candidate_paths: string[];
};

export type RegistrationPreflightItemDto = {
  path: string;
  class: string;
  reason: string | null;
};

export type RegistrationPreflightResponse = {
  preflight_id: string;
  created_at: string;
  items: RegistrationPreflightItemDto[];
};

export type RegistrationSubmitRequest = {
  aggregate_root: string;
  preflight_id: string;
  confirmed_paths: string[];
};

export type RegistrationBatchItemDto = {
  path: string;
  status: string;
  failure_reason: string | null;
};

export type RegistrationBatchDto = {
  batch_id: string;
  status: string;
  items: RegistrationBatchItemDto[];
};

File: src/web/types/logical_codebase_registration.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationPreflightRequest {
    pub aggregate_root: String,
    pub candidate_paths: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationPreflightItemDto {
    pub path: String,
    pub class: String,
    pub reason: Option<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationPreflightResponse {
    pub preflight_id: String,
    pub created_at: String,
    pub items: Vec<RegistrationPreflightItemDto>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationSubmitRequest {
    pub aggregate_root: String,
    pub preflight_id: String,
    pub confirmed_paths: Vec<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationBatchItemDto {
    pub path: String,
    pub status: String,
    pub failure_reason: Option<String>,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub struct RegistrationBatchDto {
    pub batch_id: String,
    pub status: String,
    pub items: Vec<RegistrationBatchItemDto>,
}
